src/main/java/org/opensearch/security/tools/democonfig/Installer.java

```diff
@@ -0,0 +1,377 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.tools.democonfig;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.PosixFilePermission;
+import java.util.HashSet;
+import java.util.Scanner;
+import java.util.Set;
+
+/**
+ * This class installs demo configuration for security plugin
+ */
+public class Installer {
+
+    static boolean assumeyes = false;
+    static boolean initsecurity = false;
+    static boolean cluster_mode = false;
+    static int skip_updates = -1;
+    static String SCRIPT_DIR;
+    static String BASE_DIR;
+    static String OPENSEARCH_CONF_FILE;
+    static String OPENSEARCH_BIN_DIR;
+    static String OPENSEARCH_PLUGINS_DIR;
+    static String OPENSEARCH_LIB_PATH;
+    static String OPENSEARCH_INSTALL_TYPE;
+    static String OPENSEARCH_CONF_DIR;
+    static String OPENSEARCH_VERSION;
+    static String SECURITY_VERSION;
+
+    static ExecutionEnvironment environment = ExecutionEnvironment.demo;
+
+    static final String OS = System.getProperty("os.name") + " " + System.getProperty("os.version") + " " + System.getProperty("os.arch");
+
+    static final String FILE_EXTENSION = OS.toLowerCase().contains("win") ? ".bat" : ".sh";
+
+    static final String SYSTEM_INDICES = ".plugins-ml-config, .plugins-ml-connector, .plugins-ml-model-group, .plugins-ml-model, "
```

Comment:
What do you think about finding a way to standardize this to a single list somewhere and accessing it that way? 

If I remember, every time another plugin wants to add a system index they have to register it with us and we have to have them add it in multiple places. Since this is in Java now, it should be pretty straightforward to have a single global list which all operations relating to system indices can reference.

Reply:
Thoughts on addressing this in a follow-up PR? The purpose of this list currently is to write the same to opensearch.yml. If someone were to update the list they would have to modify opensearch.yml, in which case, having a static constant wouldn't matter.

But, we can figure out a way to add it as dynamic setting in `config.yml` however that is out of scope of this PR.

Reply:
Sounds good :)